target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-deco-m9plus-v2.dtsi

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2015, The Linux Foundation. All rights reserved.
+// Copyright (c) 2021, Frank Veltmans <veltmans.frank@gmail.com>
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+#include <dt-bindings/leds/common.h>
+/ {
+	aliases {
+		led-boot = &led_blue;
+		led-failsafe = &led_red;
+		led-running = &led_green;
+		led-upgrade = &led_red;
+		label-mac-device = &swport4;
+	};
+	keys {
+		compatible = "gpio-keys";
+		reset {
+			label = "reset";
+			gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+	soc {
+		tcsr@194b000 {
+			compatible = "qcom,tcsr";
+			reg = <0x194b000 0x100>;
+			qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+			status = "okay";
+		};
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+		ess_tcsr@1953000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+	};
+};
+&tlmm {
+	serial_0_pins: serial_0_pinmux {
+		pinmux {
+			pins = "gpio16", "gpio17";
+			function = "blsp_uart0";
+			bias-disable;
+		};
+	};
+	serial_1_pins: serial_1_pinmux {
+		pinmux {
+			pins = "gpio8", "gpio9", "gpio10", "gpio11";
+			function = "blsp_uart1";
+			bias-disable;
+		};
+	};
+	spi_0_pins: spi_0_pinmux {
+		pinmux {
+			pins = "gpio13", "gpio14", "gpio15";
+			function = "blsp_spi0";
+			drive-strength = <12>;
+			bias-disable;
+		};
+		pinmux_cs {
+			pins = "gpio12";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+	};
+	i2c_0_pins: i2c_0_pinmux {
+		pinmux {
+			pins = "gpio20", "gpio21";
+			function = "blsp_i2c0";
+			drive-strength = <16>;
+			bias-disable;
+		};
+	};
+	wifi_pins: wifi_pinmux {
+		pinmux_1 {
+			pins = "gpio37";
+			function = "gpio";
+			drive-strength = <6>;
+			bias-pull-up;
+			output-high;
+		};
+		pinmux_2 {
+			pins = "gpio42";
+			function = "gpio";
+			drive-strength = <6>;
+			bias-pull-up;
+			output-high;
+		};
+		pinmux_3 {
+			pins = "gpio43";
+			function = "gpio";
+			drive-strength = <6>;
+			bias-pull-up;
+		};
+	};
+};
+&blsp_dma {
+	status = "okay";
+};
+&blsp1_i2c3 {
+	pinctrl-0 = <&i2c_0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+&blsp1_uart1 {
+	pinctrl-0 = <&serial_0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+&blsp1_uart2 {
+	pinctrl-0 = <&serial_1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+&blsp1_spi1 {
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	cs-gpios = <&tlmm 12 GPIO_ACTIVE_HIGH>;
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <24000000>;
+		partitions {
```

Comment:
need to update these partitions to match layout in stock logs:
[ 1.351499] Creating 13 MTD partitions on "7980000.qcom,nand":
[ 1.357336] 0x000000000000-0x000000100000 : "0:SBL1"
[ 1.364156] 0x000000100000-0x000000200000 : "0:MIBIB"
[ 1.369913] 0x000000200000-0x000000300000 : "0:BOOTCONFIG"
[ 1.376157] 0x000000300000-0x000000400000 : "0:QSEE"
[ 1.381838] 0x000000400000-0x000000480000 : "0:CDT"
[ 1.387130] 0x000000480000-0x000000500000 : "0:BOOTCONFIG1"
[ 1.393106] 0x000000500000-0x000000680000 : "0:APPSBLENV"
[ 1.399617] 0x000000680000-0x000000880000 : "0:APPSBL"
[ 1.406247] 0x000000880000-0x000000900000 : "0:ART"
[ 1.411535] 0x000000900000-0x000003600000 : "rootfs"
[ 1.448375] mtd: device 9 (rootfs) set to be root filesystem
[ 1.453029] 0x000003600000-0x000006300000 : "rootfs_1"
[ 1.490753] 0x000006300000-0x000006c00000 : "factory_data"
[ 1.502467] 0x000006c00000-0x000007d00000 : "runtime_data"
[ 1.519875] msm_nand_probe: NANDc phys addr 0x7980000, BAM phys addr 0x7984000, BAM IRQ 133
[ 1.527218] msm_nand_probe: Allocated DMA buffer at virt_addr 0xe0a7b000, phys_addr 0x9ec66000